### ISSUE TITLE
winget fails on Windows

### ISSUE THREAD

**Issue by openHawkes** [extrawurst/gitui#1916]
**Describe the bug**
```
C:\some\path>winget install gitui
The `msstore` source requires that you view the following agreements before using.
Terms of Transaction: https://aka.ms/microsoft-store-terms-of-transaction
The source requires the current machine's 2-letter geographic region to be sent to the backend service to function properly (ex. "US").

Do you agree to all the source agreements terms?
[Y] Yes  [N] No: Y
Failed when searching source: winget
An unexpected error occurred while executing the command:
0x8a15000f : Data required by the source is missing

No packages were found among the working sources.
```

**To Reproduce**

Looking at [these instructions](https://github.com/extrawurst/gitui#various-package-managers)...

> [Winget](https://github.com/microsoft/winget-pkgs/tree/master/manifests/s/StephanDilly/gitui) (Windows)
> `winget install gitui`

1. Windows key + R
2. cmd
3. `winget install gitui`

Alternately
1. Open cmd with admin privs somehow
2. `winget install gitui`
3. See error

[Search for that error number, find nothing](https://github.com/search?q=repo%3Aextrawurst%2Fgitui%200x8a15000f%20&type=code), open issue.

**Expected behavior**
I expected things to install.

**Context (please complete the following information):**
 - Windows 11 21H2 (OS Build 22000.2416)
 - Rust version: Did I need to install this first?

**Comment by ClementNerma** [comment]
Looking at the error, seems to be a bug coming from `winget` itself, not with `gitui`.

**Comment by ClementNerma** [comment]
In the event you get error messages with codes like this, google is your friend ;)

A quick search shows this for instance : https://github.com/microsoft/winget-cli/issues/1348

**Comment by openHawkes** [comment]
I'm afraid google wasn't my friend. As I mentioned in the initial report, I did try to search, and gave the link I used. Looks like this issue should have been there then, but github's search results page, um, doesn't have the best UI for issues (why wouldn't issue be first-class search results?) My bad. 

![image](https://github.com/extrawurst/gitui/assets/104998366/c5705c9c-53dc-4464-87c2-8d56172c83db)

That said, I'm not sure I follow the resolution to that thread. Did you?

> Experiencing this issue now. Found the root cause to be that https://winget.azureedge.net/cache/source.msix is returning a 404 error. A 404 error should be communicated to the user in a more direct way than a generic "data is missing" message.

That and similar make it sound like it could not, as I guessed, find "whatever package was needed to install gitui", but I think you're right that it's winget's catalog's problems, not a missing URL or deployment issue from gitui.

But it also sounds ("_This is the first time I've ever seen winget upgrade --all function properly. What a day to be alive. I've always wanted a Windows package manager and this give me hope that it could be a stable thing some day._") like maybe winget isn't the most reliable deployment tool. I'd probably add a warning ("winget is temperamental; wait and try again" equivalent) to the gitui install steps. $0.02 YMMV.
